certbot/main.py

```diff
@@ -1242,6 +1244,16 @@ def renew(config, unused_plugins):
     :rtype: None
 
     """
+    if not sys.stdin.isatty():
```

Comment:
It's probably safer to put this down in `main` the function, so we catch any subcommand that people might have put in their crontab.

Reply:
Per our chat discussion, maybe we can find out in Sumologic whether there are any load spikes that can be attributed to uses of `certonly` (I suspect that they're all due to uses `renew`). In our chat I pointed out that there might be other unattended uses for initial issuance (if people wrap Certbot with their own scripts) and so we might not want to introduce this delay across-the-board in order to avoid slowing down random scripted uses of Certbot that aren't related to renewals.

I mean, an extreme example could be if someone had set up some kind of unattended script that calls `certbot certificates` in order to feed some sort of status monitor—it might be annoying to have this command take several minutes to execute! And since it doesn't interact with the CA at all, there's no resource-related reason to make it delay before running.

Reply:
> I mean, an extreme example could be if someone had set up some kind of unattended script that calls `certbot certificates` in order to feed some sort of status monitor—it might be annoying to have this command take several minutes to execute! And since it doesn't interact with the CA at all, there's no resource-related reason to make it delay before running.

It _does_ hit OCSP. (And, for certbot-auto, PyPI.)

Reply:
After checking the logs, looks like most of the spikiness comes from `renew`s (not all! but a lot of it) so I'm cool keeping it here.